### PR TITLE
fix(ascend): honor nodeConfig.VDeviceCount in VDeviceCount()

### PR DESCRIPTION
**Pull request by Wangmin362**
## What this PR does

`VDeviceCount()` always derived the per-card vDevice count from the global config (`MemoryAllocatable / smallest template`), **ignoring the per-node override** loaded into `am.nodeConfig` — even though the sibling method `IsHamiVnpuCore()` already prefers `am.nodeConfig`. As a result, setting `vDeviceCount` in the node config had no effect: every card kept advertising the global default slice count.

This PR makes `VDeviceCount()` prefer `nodeConfig.VDeviceCount` when present and positive, mirroring `IsHamiVnpuCore()`, and otherwise falls back to the existing global default.

```go
func (am *AscendManager) VDeviceCount() int {
	if am.nodeConfig != nil && am.nodeConfig.VDeviceCount > 0 {
		return am.nodeConfig.VDeviceCount
	}
	if len(am.config.Templates) == 0 {
		return 1
	}
	return int(am.config.MemoryAllocatable / am.config.Templates[0].Memory)
}
```

> Note: the per-node config is only loaded when the plugin is started with `--node_config_file`. The shipped `ascend-device-plugin.yaml` mounts the node-config but does not pass that flag; that is fixed separately in #99. This PR is the code side — it makes the loaded value actually take effect.

## Unit test

Added a table-driven test `TestVDeviceCount` in `internal/manager/manager_test.go` covering: no node config (global default), `VDeviceCount=0` (fallback), `VDeviceCount>0` (honored + tracks value), and the no-template edge cases.

The test **fails on the previous code and passes with this change** — proving it pins the bug:

```
# against the OLD code (before this fix):
--- FAIL: TestVDeviceCount
    --- FAIL: TestVDeviceCount/node_config_VDeviceCount=8_->_honored_(the_fix)
        manager_test.go:96: VDeviceCount() = 4, want 8
    --- FAIL: TestVDeviceCount/node_config_VDeviceCount=2_->_honored,_tracks_the_value
        manager_test.go:96: VDeviceCount() = 4, want 2
    --- FAIL: TestVDeviceCount/no_templates_but_node_override_present_->_node_override_wins
        manager_test.go:96: VDeviceCount() = 1, want 5

# with this fix:
ok  	github.com/Project-HAMi/ascend-device-plugin/internal/manager	0.002s
--- PASS: TestVDeviceCount (0.00s)  [all 6 sub-cases PASS]
```

## Real-hardware verification (8× Ascend 910B4)

Built this branch into an arm64 image and ran it on a real node with **8× Ascend 910B4** (driver 25.5.1, k8s v1.29.15, containerd 1.7.13). To avoid disturbing the node's existing device plugin, the test plugin used a **distinct `commonWord`/`resourceName`** (`huawei.com/Ascend910B4-f2test`) — a fully isolated, additive registration.

The `910B4` config has `memoryAllocatable=32768` and smallest template `8192`, so the global default is `32768/8192 = 4` vDevices/card. Evidence, per tier — the per-card `count` comes from the plugin's own HAMi node-register annotation, cross-checked against the kubelet-advertised resource capacity (8 cards):

| node config | plugin log | per-card `count` (register annotation) | node capacity (8 cards) |
|---|---|---|---|
| *(vDeviceCount unset)* | `matched node config … VDeviceCount:0` | `[4,4,4,4,4,4,4,4]` | `32` |
| `vDeviceCount: 8` | `matched node config … VDeviceCount:8` | `[8,8,8,8,8,8,8,8]` | `64` |
| `vDeviceCount: 2` | `matched node config … VDeviceCount:2` | `[2,2,2,2,2,2,2,2]` | `16` |

- **Unset → global default (4)**: unchanged behavior, no regression.
- **Set → honored and tracks the value (8→8, 2→2)** on real hardware.

All test resources were removed after the run.

## Scope / safety of the test

The real-hardware test ran as a **fully isolated, additive** deployment: a separate namespace, a distinct `commonWord`/`resourceName` (`huawei.com/Ascend910B4-f2test`) so it never collided with the production `huawei.com/Ascend910B4` registration or its HAMi node annotations. All test resources were removed afterward and the node was restored to its exact prior state.

---

*AI assistance disclosure (per the project's AI Assistance Notice): the underlying issue was diagnosed from my own testing on real Ascend 910B4 hardware. The one-function change and the unit test were drafted with AI assistance (Claude Code) and then reviewed, built, and verified by me on the hardware described above. I understand the change and stand behind it.*

## Summary by CodeRabbit

* **Bug Fixes**
  * `VDeviceCount()` now honors a per-node vDevice override when a node configuration is present and the override is greater than zero.
  * When no valid override applies (including missing templates), it falls back to the computed default, using a safe value of `1` as needed.

* **Tests**
  * Added unit test coverage for `VDeviceCount()` across default, node-overridden, zero-value override, and missing-template scenarios.

### CODE REVIEW

File: internal/manager/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 The HAMi Authors.
```
Comment (DSFans2014) [inline internal/manager/manager_test.go]: 2026

Comment (Wangmin362) [reply internal/manager/manager_test.go]: Thanks — fixed, updated the header to `Copyright 2026` in 1ffd7f9.

**Review by DSFans2014 — COMMENTED** [review]
/lgtm

**Review by DSFans2014 — COMMENTED** [review]
/lgtm

**Review by archlitchi — APPROVED** [review]
/lgtm
